### PR TITLE
Check if route exist before checking Content-Type of body

### PR DESCRIPTION
**Pull request by mage1k99**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
Cant we check before we run the content type parser if we havea 404?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mage1k99** [comment]
> Cant we check before we run the content type parser if we havea 404?

I check the if it's a 404 before checking Content-Type. so that it does not raise 415 on a 404 route just because it had a request body

**Comment by mage1k99** [comment]
I have updated the check. thanks to @Uzlopak for giving me hint

All the tests now passes

File: lib/contentTypeParser.js [hunk 1]
```diff
@@ -149,7 +149,11 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
   const resource = new AsyncResource('content-type-parser:run', request)
 
   if (parser === undefined) {
-    reply.send(new FST_ERR_CTP_INVALID_MEDIA_TYPE(contentType || undefined))
+    if (request.is404) {
+      return handler(request, reply)
```
Comment (Eomm) [inline lib/contentTypeParser.js:153]: just to avoid returning something 

```suggestion
     handler(request, reply)
     return
```

**Review by Eomm — APPROVED** [review]

**Pull request merged**